application-analytics-test/application-analytics-test-pageobjects/src/main/java/com/xwiki/analytics/test/po/AdminViewPage.java

```diff
@@ -0,0 +1,123 @@
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+import org.xwiki.administration.test.po.AdministrationPage;
+import org.xwiki.test.ui.XWikiWebDriver;
+import org.xwiki.test.ui.po.ViewPage;
+
+public class AdminViewPage extends ViewPage
+{
+    private static final String trackingCodeId = "Analytics.Code.ConfigurationClass_0_trackingCode";
+
+    private static final String authTokenId = "Analytics.Code.ConfigurationClass_0_authToken";
+
+    private static final String idSiteId = "Analytics.Code.ConfigurationClass_0_siteId";
+
+    private static final String requestAddressId = "Analytics.Code.ConfigurationClass_0_requestAddress";
+
+    private static XWikiWebDriver driver;
+    public AdminViewPage()
+    {
+            driver = getUtil().getDriver();
+    }
+
+    public static AdminViewPage gotoAdminPage()
+    {
+        AdministrationPage administrationPage = AdministrationPage.gotoPage();
+        administrationPage.clickSection("Other", "Analytics");
+        return new AdminViewPage();
+    }
+
+    public AdminViewPage setTrackingCode(String value)
+    {
+        WebElement element = driver.findElement(By.id(trackingCodeId));
+        element.clear();
+        element.sendKeys(value);
+        return this;
+    }
+
+    public AdminViewPage setAuthTokenId( String value)
+    {
+        WebElement element = driver.findElement(By.id(authTokenId));
+        element.clear();
+        element.sendKeys(value);
+        return this;
+    }
+
+    public AdminViewPage setIdSiteId( String value)
+    {
+        WebElement element = driver.findElement(By.id(idSiteId));
+        element.clear();
+        element.sendKeys(value);
+        return this;
+    }
+
+    public AdminViewPage setRequestAddressId(String value)
+    {
+        WebElement element = driver.findElement(By.id(requestAddressId));
+        element.clear();
+        element.sendKeys(value);
+        return this;
+    }
+
+    public AdminViewPage bringSaveButtonIntoView()
+    {
+        WebElement saveButton = driver.findElement(By.cssSelector(".btn.btn-primary"));
+        Actions actions = driver.createActions();
+        actions.moveToElement(saveButton).click().perform();
+        return this;
+    }
+
+    public boolean inProgressNotification(String message)
+    {
+
+        try {
+            this.waitForNotificationInProgressMessage(message);
+            return true;
+        } catch (Exception e) {
+            return false;
+        }
+    }
+
+    public boolean errorNotification(String message)
+    {
+
+        try {
+            this.waitForNotificationErrorMessage(message);
+            return true;
+        } catch (Exception e) {
+            return false;
+        }
+    }
+
+    public boolean successNotification(String message)
+    {
+        try {
+            this.waitForNotificationSuccessMessage(message);
+            return true;
+        } catch (Exception e) {
+            return false;
+        }
+    }
```

Comment:
I don't understand the need for these methods. If the wait throws an exception then the test fails. Why write

     assertTrue(adminViewPage.successNotification("Saved"));

when you can simply write:

    adminViewPage.waitForNotificationSuccessMessage("Saved");

The test will continue if the wait is successful, otherwise it will fail with a timeout.